packages/itwinui-react/src/utils/components/OverflowContainer.tsx

```diff
@@ -370,51 +382,51 @@ const overflowGuessReducer = (
 ): GuessState => {
   /** Ensure that the visibleCount is always ≤ itemsCount */
   const getSafeVisibleCount = (visibleCount: number) =>
-    Math.min(state.itemsCount, visibleCount);
+    Math.min(action.state.itemsCount, visibleCount);
 
   switch (action.type) {
     case 'decreaseMaxGuess':
     case 'increaseMinGuess':
-      if (state.isStabilized) {
-        return state;
+      if (action.state.isStabilized) {
+        return action.state;
       }
 
-      let newMinGuess = state.minGuess;
-      let newMaxGuess = state.maxGuess;
+      let newMinGuess = action.state.minGuess;
+      let newMaxGuess = action.state.maxGuess;
```

Comment:
I'll repeat what I said internally, which is that we should only use `action.state` should only be used where it is actually necessary, rather than replacing all instances of `state`.

Also, a more descriptive name would help distinguish it from `state`.

Reply:
> Since we're using `action.state` in some places, I thought it could lead to two sources of truth and confusion. So, I thought to just use one everywhere.

It's more confusing to use `action.state` because that's not a thing that's usually done. `state` is the more familiar pattern.

Wherever we use `action.state`, we should add a code comment explaining why. This way, we avoid doing the unusual thing where unnecessary and address any possible confusion at the same time.

And since we do not need it for all actions, we can also avoid passing it.

> Thus, I don't think `action.state` is the "previous state."

It is still the "previous state", because the reducer is used for calculating the "next state". When using reducers, an action is often dispatched with information about the "next"/"desired" state, rather than the previous/current one.

You could name it `currentState` if that's more clear to you. The actual name is not super important, I just want to meaningfully differentiate it from the `state` argument. But this is the less important part of the comment. I think code comments would be helpful regardless of the name.

Reply:
Yeah, my thinking is that any workaround should be kept confined to the minimum required surface area.

Investigating it in the future sounds good. There is a related TODO in #2154 which could be investigated at the same time (we did technically reduce the number of dependencies, but by relying on `guessState` we actually re-execute the Effect even more frequently, so the TODO is still unsolved).